libs/src/utils/multiProvider.js

```diff
@@ -6,24 +6,24 @@ const { shuffle } = require('lodash')
  * web3 consumes a provider object on initialization
  * ref: https://github.com/ChainSafe/web3.js/blob/1.x/packages/web3/types/index.d.ts#L31
  * which references: https://github.com/ChainSafe/web3.js/blob/1.x/packages/web3-core/types/index.d.ts#L436
- * MultiProivder implements HttpProvider which can be consumed by web3
+ * MultiProvider implements HttpProvider which can be consumed by web3
  * ref for HttpProvider: https://github.com/ChainSafe/web3.js/blob/1.x/packages/web3-providers-http/types/index.d.ts#L46-L66
  */
 class MultiProvider extends Web3.providers.HttpProvider {
   /**
    * Creates a MultiProvider
-   * @param {Array<String | Provider>} providers - The providers to use.
+   * @param {Array<string | Provider> | string} - The providers to use.
    */
   constructor (providers) {
-    super()
-
     if (typeof providers === 'string') {
       providers = providers.split(',')
     } else if (!Array.isArray(providers)) {
       providers = [providers]
     }
 
+    // The below line ensures that we support different types of providers i.e. comma separated strings, an array of strings or an array of providers.
     providers = providers.map(provider => (new Web3(provider)).eth.currentProvider)
+    super(providers[0].host)
```

Comment:
this looks good, assuming you actually confirmed that the `host` property is what you expect it to be
the other option would be to just call `super(providers)` on line 18 like you had originally. that seems like simpler option to me but i guess either work

Reply:
This can't be done on line 18, since providers could be a a type that is not supported by HttpProvider. As HttpProvider expects a single provider string.

Reply:
ah right, of course. mixed it up with the web3 constructor pattern haha, sounds great 👍